### DISCUSSION TITLE
Add multi-language support

### DISCUSSION THREAD

**Discussion by lony2003** [authorizerdev/authorizer#438] [Ideas and features]
Hi! I'm a user and developer from China and I want to translate this program into Chinese. Have you schedued to implement the multi-language support? If not, I would like to give some help.

**Comment by lakhansamani** [comment]
@lony2003 some help would be really helpful
Need to work on translation model.
I can write a detailed RFC for that if you like

**Comment by lony2003** [comment]
@lakhansamani Well, that would be greate to me. In my conception, I would like to use [react-intl-universal](https://github.com/alibaba/react-intl-universal) to achive that since react-intl is huge and we just need some features of it.

To make the whole process seamless, I think I'll make this process into three parts.

- Import the libraries and replace strings in this program. All the strings in the program will be replaced to this format like 
```javascript
intl.get('HELLO', { name }).d(`Hello, ${name}`);
```

- Generate the messages list with `react-intl-universal-extract`, a tool from `react-intl-universal`. It will generate a json file like 
```json
{
  "hello1": "Hello World",
  "hello2": "Hello {name}",
}
```
and you can upload it to some localization platform like `weblate` and translate it. Then get the translated json from it.

- Add some `select language` and `help us translate` stuff in pages.

This process can solve most of the problem except "RTL" direction of the page, but it can be implement later.
